src/transformers/models/cpmant/modeling_cpmant.py

```diff
@@ -0,0 +1,1090 @@
+
+    @add_start_docstrings_to_model_forward(CPMANT_INPUTS_DOCSTRING)
+    @add_code_sample_docstrings(
+        processor_class=_TOKENIZER_FOR_DOC,
+        checkpoint=_CHECKPOINT_FOR_DOC,
+        output_type=BaseModelOutput,
+        config_class=_CONFIG_FOR_DOC,
+    )
+    def forward(
+        self,
+        input_ids: Optional[torch.Tensor] = None,
+        length: Optional[torch.Tensor] = None,
+        context: Optional[torch.Tensor] = None,
+        position: Optional[torch.Tensor] = None,
+        segment: Optional[torch.Tensor] = None,
+        span: Optional[torch.Tensor] = None,
+        return_dict: Optional[bool] = False,
+        **kwargs,
+    ):
+        r"""
+        Args:
+            input_ids (`torch.Tensor` of shape `(batch_size, seq_len)`):
+                Indices of input sequence tokens in the vocabulary.
+
+                Indices can be obtained using [`CPMAntTokenizer`]. See [`PreTrainedTokenizer.encode`] and
+                [`PreTrainedTokenizer.__call__`] for details.
+
+                [What are input IDs?](../glossary#input-ids)
+
+            length (`torch.Tensor` of shape `(batch)`, *optional*):
+                The length of input tokens.
+            context (`torch.Tensor` of shape `(batch_size, seq_len)`, *optional*):
+                The Boolean value determines whether the model makes a prediction for that position
+            position (`torch.Tensor` of shape `(batch_size, seq_len)`, *optional*):
+                Indices of position of each input sequence tokens in the position embeddings. Selected in the range
+                `[0, config.max_position_embeddings - 1]`.
+
+                [What are position IDs?](../glossary#position-ids)
+            segment (`torch.Tensor` of shape `(batch_size, seq_len)`, *optional*):
+                A sequence of tokens that is processed together as a unit.
+            span (`torch.Tensor` of shape `(batch_size, seq_len)`, *optional*):
+                A contiguous sequence of tokens within the input text.
+            return_dict (`bool`, *optional*):
+                Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+        """
+        return_dict = return_dict if return_dict is not None else self.config.return_dict
+        input_ids = input_ids.contiguous()
+        hidden_states = self.input_embedding(input_ids)
+        segment_states = self.segment_embedding(segment)
+        hidden_states = hidden_states + segment_states
+
+        attention_mask = self._prepare_attention_mask(input_ids, span, context, length)
+        position_bias = self.position_bias(position, position, segment, segment)
+
+        hidden_states = self.encoder(hidden_states, attention_mask, position_bias)
+        logits = F.linear(hidden_states, self.input_embedding.weight)
+
+        if not return_dict:
+            return tuple(v for v in [logits, hidden_states] if v is not None)
+
+        return BaseModelOutput(hidden_states=hidden_states)
```

Comment:
some attributes are missing such as the attention outputs etc.

Reply:
We added some attributes.